src/index.ts

```diff
@@ -1,10 +1,28 @@
 import { Box, Quaternion as CQuaternion, ConvexPolyhedron, Cylinder, Shape, Sphere, Trimesh, Vec3 } from 'cannon-es';
 import { Box3, BufferGeometry, CylinderGeometry, MathUtils, Mesh, Object3D, SphereGeometry, Vector3 } from 'three';
 import { ConvexHull } from '../lib/ConvexHull.js';
-import { getComponent, getGeometry, getVertices } from './utils.js';
+import { getComponent, getGeometry, getVertices } from './utils';
 
 const PI_2 = Math.PI / 2;
 
+export type BoxParameters = [x: number, y: number, z: number];
+
+export type CylinderParameters = [radiusTop: number, radiusBottom: number, height: number, segements: number];
+
+export type SphereParameters = number;
+
+export type ConvexPolyhedronParameters = { vertices: [number, number, number][], faces: number[][] };
+
+export type TrimeshParameters = [vertices: number[], indices: number[]];
```

Comment:
Good question...

I made some dodgy benchmarks here: https://codesandbox.io/s/web-worker-array-vs-transferable-tests-jqohwl

Looks like the cost of creating a Float32Array is notably higher than creating a regular array or a nested array e.g. `[number, number, number][]`). Unless I am doing something wrong!

If that's the case, maybe it's best to just return regular arrays. Then if someone wants to send data to/from a worker often and get the performance boost from a transferable array, they can create it themselves?

Otherwise, we'll be pushing the cost to people using `threeToCannon`, who won't know or care that a Float32Array was created internally.

Have a look and let me know what you think! 🙂

Reply:
It still looks like it's faster to create regular and nested arrays (`number[]` or `[number, number, number][]`, but it's less notable.

With `N = 10000000`, the times look like this:
```
"create regular array" = 14.70000ms 
"create nested array" = 13.70000ms 
"create float32array and set values in a for-loop" = 27.60000ms 
```

Reply:
It also looks like creating a nested array as-is in this PR is not so performant

with `N = 10000000`

creating a regular array takes ~13.90000ms
creating a nested array takes ~115.20000ms
creating a float32array and setting values in a for loop takes ~27.00000ms

Reply:
So with the above in mind:

- If want the fastest option, we can just return a flat array
- If we want an API that mirrors cannon's more closely, we can return a nested array in `ConvexPolyhedronParameters` and a flat array in `TrimeshParameters`
- If we don't mind pushing the slight added cost of creating a Float32Array to people just using `threeToCannon`, we can return a Float32Array for both

What are your thoughts?

Reply:
This shouldn't sway the decision, as other users should be considered -

But for use-cannon, some refactoring would be required to send trimesh & convex shape params as transferable float32s. At least for the initial implementation of the shape inference hook, I probably won't do that refactoring.

Reply:
The main advantage of typed arrays would be that we can send them to/from a Web Worker without copying data, they're [transferable](https://developer.mozilla.org/en-US/docs/Glossary/Transferable_objects). Same for upload to GPU if needed. For me that would outweigh the small difference in write time between number[] and float32array.

I guess the only issue is that ConvexPolyhedronParameters may contain n-gons, so we need at least one nested array there? In that case I'd vote for:

```js
export type ConvexPolyhedronParameters = { vertices: Float32Array, faces: number[][] };

export type TrimeshParameters = { vertices: Float32Array, indices: Uint32Array };
```

Does that sound ok?

Reply:
That sounds good!

I'll update this PR soon 🙂

Reply:
Updated now! @donmccurdy 

There's one thing that could be improved. In `getConvexPolyhedronParameters` I'm building a regular array for vertices, then creating a typed array from that.

If there's a good way of knowing how many vertices to pre-allocate for, this can be changed to just populating the typed array. The `ConvexHull` class _does_ have a `vertices` property, but it doesn't have the same length as the vertices array built from iterating over face edges.

Reply:
Not too worried about it – there's a lot of lower-hanging optimization left to be had in the library. Thanks!